# Generated by Django 4.2.19 on 2025-03-03 09:55

import django.db.models.deletion
from django.conf import settings
from django.db import migrations, models


class Migration(migrations.Migration):

    dependencies = [
        migrations.swappable_dependency(settings.AUTH_USER_MODEL),
        ('recovery', '0019_auto_20250111_1343'),
    ]

    operations = [
        migrations.RemoveField(
            model_name='recoverycurrency',
            name='contract',
        ),
        migrations.AddField(
            model_name='recoverynetwork',
            name='url_address',
            field=models.URLField(null=True),
        ),
        migrations.AddField(
            model_name='recoverynetwork',
            name='url_hash',
            field=models.URLField(null=True),
        ),
        migrations.CreateModel(
            name='RejectReason',
            fields=[
                ('id', models.AutoField(auto_created=True, primary_key=True, serialize=False, verbose_name='ID')),
                ('title', models.CharField(max_length=35, unique=True)),
                ('description', models.TextField(null=True)),
                ('created_at', models.DateTimeField(auto_now_add=True)),
                (
                    'allocated_by',
                    models.ForeignKey(
                        null=True,
                        on_delete=django.db.models.deletion.SET_NULL,
                        to=settings.AUTH_USER_MODEL,
                        verbose_name='ایجاد کننده',
                    ),
                ),
            ],
        ),
        migrations.CreateModel(
            name='RecoveryRejectReasons',
            fields=[
                ('id', models.AutoField(auto_created=True, primary_key=True, serialize=False, verbose_name='ID')),
                ('created_at', models.DateTimeField(auto_now_add=True)),
                (
                    'allocated_by',
                    models.ForeignKey(
                        null=True,
                        on_delete=django.db.models.deletion.SET_NULL,
                        to=settings.AUTH_USER_MODEL,
                        verbose_name='ایجاد کننده',
                    ),
                ),
                (
                    'recovery',
                    models.ForeignKey(
                        null=True,
                        on_delete=django.db.models.deletion.SET_NULL,
                        related_name='reject_reason',
                        to='recovery.recoveryrequest',
                    ),
                ),
                ('reseans', models.ManyToManyField(to='recovery.rejectreason')),
            ],
        ),
    ]
